### PR TITLE
Fix duplication not working as expected on grabbed entities

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11498/HighFidelity-Beta-PR13671-182a90d.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11498/HighFidelity-Beta-PR13671-182a90d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11498/HighFidelity-Beta-Interface-PR13671-182a90d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11498/HighFidelity-Beta-PR13671-182a90d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11498/HighFidelity-Beta-Interface-PR13671-182a90d.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11629/HighFidelity-Beta-PR13671-103f9e0.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11629/HighFidelity-Beta-PR13671-103f9e0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11629/HighFidelity-Beta-Interface-PR13671-103f9e0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11629/HighFidelity-Beta-PR13671-103f9e0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11629/HighFidelity-Beta-Interface-PR13671-103f9e0.exe

**Comment by huffman** [comment]
https://highfidelity.manuscript.com/f/cases/2807/Cube-copied-while-being-held-acts-odd-and-follows-avatar

You will need 2 users. User 1 should be in Desktop, the other should be in an HMD with hand controllers.

# Test Plan

## Part 1
1) User 1: Open Create and create a Cube entity. Make Cube dynamic
2) User 2: Near grab previously created cube
3) User 1: While User 2 has the previously created cube near grabbed, copy the dynamic cube entity (With create still open and cube in focus, Alt + Left click cube)
4) User 2: Without letting go of the original cube, move around
5) User 1 and 2 should see the original cube following User 2, while the duplicated cube does not move.  **Previous behavior:** Duplicate cube follows User 2 around, original cube is still near grabbed by User 2, User 1 does NOT see duplicate cube following User 2.
6) User 2: Release near grab of original cube, and move around
7) User 1 and 2 should see both cubes not moving. **Previous Behavior:** Original cube follows User 2 while they move around, duplicate cube does not follow User 2 anymore, User 1 does NOT see cube following User 2 until the cube is interacted with.

## Part 2
Run through the steps in Part 1 again but instead of having User 1 near grab the cube, they should far-grab the cube.

## Part 3
Run through the steps in Part 1 again with a cube that is **not** dynamic.

## Part 4
From the Running Scripts window, run `developer > tests > dynamics > dynamicsTests.js`. Open the Dynamics window by pressing the `Dynamics` button in the toolbar. Click the `Ragdoll` button to create a ragdoll. Run through the steps 2-7 in Part 1 again, but User 2 should grab the Ragdoll torso.

**Comment by qliejahner** [comment]
Currently testing

**Comment by qliejahner** [comment]
FAIL @conklin94122

Unexpected result for ragdoll: New rectangle is attached to the ragdoll and causes erratic movement (with or without User 2 grabbing it). Does not occur on master (9125)

hifi-snap-by--on-2018-07-27_16-18-45
[hifi-log (1).txt](https://github.com/highfidelity/hifi/files/2237418/hifi-log.1.txt)

**Comment by huffman** [comment]
Hi @qliejahner ! This is my fault, I should have been more clear. Although the results are disturbing in this case (😄) this is the expected behavior, assuming that neither entity remained stuck to User 2. The intention was to maintain the same physics actions / relations on the duplicated entity.

Were the other tests successful?

**Comment by qliejahner** [comment]
Sorry for not specifying while leaving the result, but that was the only part of the test that looked like it might be a problem.

PASS @conklin94122 

Duplicated entities were not affected by User 2's grasp on the original entity.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11719/HighFidelity-Beta-PR13671-ddb900a.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11719/HighFidelity-Beta-PR13671-ddb900a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11719/HighFidelity-Beta-Interface-PR13671-ddb900a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11719/HighFidelity-Beta-PR13671-ddb900a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11719/HighFidelity-Beta-Interface-PR13671-ddb900a.exe

**Comment by qlilaseguinolaza** [comment]
Currently testing

**Comment by qlilaseguinolaza** [comment]
Currently testing

**Comment by qlilaseguinolaza** [comment]
Stopped testing 

CREATE app missing
image

File: scripts/system/libraries/entitySelectionTool.js [hunk 1]
```diff
@@ -198,12 +198,12 @@ SelectionManager = (function() {
     // This is mostly a heuristic - there is no perfect way to know if an entity is being
     // grabbed.
     function nonDynamicEntityIsBeingGrabbedByAvatar(properties) {
-        if (properties.dynamic || properties.parentID === null) {
+        if (properties.dynamic || Uuid.isNull(properties.parentID) {
```
Comment (thoys) [inline scripts/system/libraries/entitySelectionTool.js]: missing `)`

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11737/HighFidelity-Beta-PR13671-bbdc7ba.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11737/HighFidelity-Beta-PR13671-bbdc7ba.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11737/HighFidelity-Beta-Interface-PR13671-bbdc7ba.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11737/HighFidelity-Beta-PR13671-bbdc7ba.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11737/HighFidelity-Beta-Interface-PR13671-bbdc7ba.exe

**Comment by QliCknox** [comment]
Currently Testing

**Comment by QliCknox** [comment]
Pass @conklin94122 

Cloned entities did not follow the user once duplicated

**Pull request merged**